test-harness/src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -746,34 +656,6 @@ public FreeStyleProject createFreeStyleProject(String name) throws IOException {
         return jenkins.createProject(FreeStyleProject.class, name);
     }
 
-    public MatrixProject createMatrixProject() throws IOException {
```

Comment:
Cannot keep this without an unwanted plugin dep. Anyway it is very simply to use `Jenkins.createProject(Class, String)` directly from a test.